packages/astro/src/vite-plugin-astro-server/route.ts

```diff
@@ -57,12 +65,18 @@ export async function matchRoute(
 			return {
 				route: maybeRoute,
 				filePath,
+				resolvedPathname: pathname,
 				preloadedComponent,
 				mod,
 			};
 		}
 	}
 
+	const altPathname = pathname.replace(/(index)?\.html$/, '');
```

Comment:
can you add some comments around this? I think I understand why this is being done but probably won't remember a month from now!

Reply:
Added comments in [d2785c3](https://github.com/withastro/astro/pull/5346/commits/d2785c3fb8d469f9958d5fdb9b44ee53e223d4d8)